### PR TITLE
[Product Blocks editor]: add tooltip help to Organization/Product catalog and Attributes

### LINKED ISSUE #39636: [Fix] Add missing description to Product catalog and Attributes
We don't have the description in the `Product catalog` and `Attributes` sections in the new `Organization` tab (when clicking on the tooltip ). Let's add it.

**Acceptance criteria**

- [ ] `Product catalog` section has a description in the tooltip that reads: `Help customers find this product by assigning it to categories, adding extra details, and managing its visibility in your store and other channels.`
- [ ] Attributes section has a description in the tooltip that reads: `Add descriptive pieces of information that customers can use to filter and search for this product. Learn more`. `Learn more` links to [documentation](https://woo.com/document/managing-product-taxonomies/#product-attributes).

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR adds descriptions (tooltip help) to the _Product catalog_ and `Attributes` section of the _Organization_ page.

Closes https://github.com/woocommerce/woocommerce/issues/39636

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Go to the Product Editor
2. Click on the _Organization_ tab
3. Confirm you see the `(?)` button next to the both Product catalog and Attributes sections
4. Confirm you see the help tooltip by clicking on the buttons:

Screenshot 2023-11-22 at 15 10 36

Screenshot 2023-11-22 at 15 10 19

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: plugins/woocommerce/src/Internal/Features/ProductBlockEditor/ProductTemplates/SimpleProductTemplate.php [hunk 1]
```diff
@@ -882,7 +884,6 @@ private function add_shipping_group_blocks() {
 						'checkedValue'   => false,
 						'uncheckedValue' => true,
 						'label'          => __( 'This product requires shipping or pickup', 'woocommerce' ),
-						'uncheckedHelp'  => __( 'This product will not trigger your customer\'s shipping calculator in cart or at checkout. This product also won\'t require your customers to enter their shipping details at checkout. <a href="https://woo.com/document/managing-products/#adding-a-virtual-product" target="_blank" rel="noreferrer">Read more about virtual products</a>.', 'woocommerce' ),
 					),
```
Comment (louwie17) [inline plugins/woocommerce/src/Internal/Features/ProductBlockEditor/ProductTemplates/SimpleProductTemplate.php]: Did you mean to remove this? This should stay shouldn't it?

**Review by louwie17 — COMMENTED** [review]
Tested well, although left a comment in relation to the removal of the `uncheckedHelp` text.

File: plugins/woocommerce/src/Internal/Features/ProductBlockEditor/ProductTemplates/SimpleProductTemplate.php (hunk 1 above)
Comment (retrofox) [reply plugins/woocommerce/src/Internal/Features/ProductBlockEditor/ProductTemplates/SimpleProductTemplate.php]: Addressed. :-(

**Review by louwie17 — APPROVED** [review]
Thanks, nice work! Change looks good and tested well 👍

**Pull request merged**